### PR TITLE
chore: update deps and types

### PR DESCRIPTION
**Pull request by achingbrain**
Updates all deps, pulls in tsconfig from aegir and fixes the new ts issues that identified.

### CODE REVIEW

**Comment by vasco-santos** [comment]
I just did the same PR first, but you added the types 😂  I will close https://github.com/multiformats/js-mafmt/pull/78

Can you just port from there the Readme change and node versions in CI?

File: src/index.js [hunk 1]
```diff
@@ -144,25 +149,44 @@ exports.IPFS = P2P
  * Validation funcs
```
Comment (vasco-santos) [inline src/index.js:149]: Can you update the above exports like in #78 ?

Comment (achingbrain) [reply src/index.js:152]: Done

**Review by vasco-santos — APPROVED** [review]
LGTM!
Thanks for adding all the types

**Pull request merged**